game/src/contrib/systems/HealthSystem.java

```diff
@@ -28,28 +29,55 @@ public HealthSystem() {
     public void execute() {
         entityStream()
                 // Consider only entities that have a HealthComponent
-                // Form triples (e, hc, ac)
+                // Form triples (e, hc, dc)
                 .map(this::buildDataObject)
                 // Apply damage
                 .map(this::applyDamage)
                 // Filter all dead entities
                 .filter(hsd -> hsd.hc.isDead())
-                .filter(
-                        hsd -> {
-                            if (!hsd.ac.hasAnimation(AdditionalAnimations.DIE)
-                                    || hsd.ac
-                                            .getAnimation(AdditionalAnimations.DIE)
-                                            .get()
-                                            .isLooping()) return true;
-                            if (!hsd.ac.isCurrentAnimation(AdditionalAnimations.DIE)) {
-                                hsd.ac.currentAnimation(AdditionalAnimations.DIE);
-                            }
-                            return hsd.ac.currentAnimation().isFinished();
-                        })
+                // Set DeathAnimation if possible and not yet set
+                .map(this::activateDeathAnimation)
+                // Filter by state of animation
+                .filter(this::testDeathAnimationStatus)
                 // Remove all dead entities
                 .forEach(this::removeDeadEntities);
     }
 
+    /**
+     * Tests the existence and current status of the DeathAnimation of an entity.
+     *
+     * <p>Returns a corresponding Boolean value if the entity can be removed from the game. This is
+     * the case if the entity does not have a death animation, or it has already finished. Also, if
+     * the entity has a death animation, and it is in loop mode, the entity will be marked for
+     * removal.
+     *
+     * @param hsd HSData to check Animations in
+     * @return true if Entity can be removed from the game
+     */
+    private boolean testDeathAnimationStatus(HSData hsd) {
+        DrawComponent dc = hsd.dc;
+        // test if hsd has a DeathAnimation
+        Predicate<DrawComponent> hasDeathAnimation =
+                (drawComponent) -> drawComponent.hasAnimation(AdditionalAnimations.DIE);
+        // test if Animation is looping
+        Predicate<DrawComponent> isAnimationLooping = DrawComponent::isCurrentAnimationLooping;
+        // test if Animation has finished playing
+        Predicate<DrawComponent> isAnimationFinished = DrawComponent::isCurrentAnimationFinished;
+
+        return !hasDeathAnimation.test(dc)
+                || isAnimationLooping.test(dc)
+                || isAnimationFinished.test(dc);
+    }
+
+    private HSData activateDeathAnimation(HSData hsd) {
+        // set DeathAnimation as active animation
+        if (!hsd.dc.isCurrentAnimation(AdditionalAnimations.DIE)) {
+            hsd.dc.currentAnimation(AdditionalAnimations.DIE);
```

Comment:
müsste nicht vorher geprüft werden, ob das Ding überhaupt eine DeathAnimation hat? 

Wenn ich den Stream richtig lese, wird erst die Animation gesetzt und dann in `testDeathAnimationStatus`, ob das Ding überhaupt eine death animation hat.